AwesomeCalculator/app/src/main/java/com/example/drcreeper/awesomecalculator/fragments/CalculatorFragment.java

```diff
@@ -0,0 +1,154 @@
+    @BindView(R.id.num_dot_button)
+    Button dotButton;
+    @BindView(R.id.add_button)
+    Button addButton;
+    @BindView(R.id.sub_button)
+    Button subButton;
+    @BindView(R.id.mul_button)
+    Button mulButton;
+    @BindView(R.id.div_button)
+    Button divButton;
+    @BindView(R.id.clear_button)
+    Button clearButton;
+    @BindView(R.id.erace_button)
+    Button eraseButton;
+    private Calculator calculator;
+
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+        View view = inflater.inflate(R.layout.calculator_fragment, container, false);
+        super.onCreate(savedInstanceState);
+        ButterKnife.bind(this, view);
+        calculator = new Calculator();
+        CalculatorWriter.restoreState(calculator,getActivity().getSharedPreferences(CalculatorPreferencesContract.FILE_NAME, Context.MODE_PRIVATE));
+        refresh();
+        for (final Button button : numbersButtons) {
+            button.setOnClickListener((v) -> {
+                    calculator.numPress(button.getText().charAt(0));
+                    refresh();
+            });
+        }
+        dotButton.setOnClickListener((v)-> {
+                calculator.numPress('.');
+                refresh();
+        });
+        addButton.setOnClickListener((v) -> {
+                calculator.operatorPress(Operator.ADD);
+                refresh();
+        });
+        subButton.setOnClickListener((v) -> {
+                calculator.operatorPress(Operator.SUB);
+                refresh();
+        });
+        mulButton.setOnClickListener((v) -> {
+                calculator.operatorPress(Operator.MUL);
+                refresh();
+        });
+        divButton.setOnClickListener((v) -> {
+                calculator.operatorPress(Operator.DIV);
+                refresh();
+        });
+
+        clearButton.setOnClickListener(this::onClearClick);
+        eraseButton.setOnClickListener(this::onEraseClick);
+        return  view;
+    }
+
+    @OnClick(R.id.slove_button) //the best
+    public void onSolveClick() {
+        calculator.solvePress();
+        refresh();
+        SaveOperationToHistoryAsyncTask writer = new SaveOperationToHistoryAsyncTask();
+        writer.setContext(getContext());
+        writer.execute(calculator.getHistoryLog());
+    }
+
+    public void onClearClick(View v) {
+        calculator.clear();
+        refresh();
+    }
+
+    public void onEraseClick(View v) {
+        calculator.erasePress();
+        refresh();
+    }
+
+    public void refresh() {
+        mainTextView.setText(calculator.getCurrentText());
+        SharedPreferences sharedPreferences = getActivity().getSharedPreferences(CalculatorPreferencesContract.FILE_NAME,0);
+        CalculatorWriter.saveState(calculator,sharedPreferences);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
```

Comment:
This is also lifecycle event and should be put before non-lifecycle methods in code of fragment